src/reactionHandlers/updateReactboard.ts

```diff
@@ -0,0 +1,188 @@
+import {
+	Attachment,
+	ChannelType,
+	DMChannel,
+	EmbedBuilder,
+	Message,
+	MessageReaction,
+	NewsChannel,
+	PartialDMChannel,
+	PrivateThreadChannel,
+	PublicThreadChannel,
+	TextChannel,
+	VoiceChannel,
+} from 'discord.js';
+import { db } from '../database';
+import { appVersion } from '../constants/meta';
+
+export const updateReactboard: ReactionHandler = {
+	async execute({ reaction, user }) {
+		const fullReaction = await reaction.fetch();
+		const fullMessage = await reaction.message.fetch();
+		const fullUser = await user.fetch();
+		if (fullMessage.guildId === null) return; // ignore guildless messages
+
+		const reactboardExists =
+			(await db.reactboard.count({
+				where: {
+					guildId: fullMessage.guildId,
+					react: getDbReactName(fullReaction),
+				},
+			})) > 0;
+		if (reactboardExists) {
+			if (fullMessage.author.bot) {
+				await fullMessage.channel.send(
+					`${user.toString()}, you can't use that react on bot messages!`
+				);
+				await reaction.users.remove(fullUser);
+				return;
+			}
+
+			if (fullMessage.author.id === user.id) {
+				await fullMessage.channel.send(
+					`${user.toString()}, you can't use that react your own messages!`
+				);
+				await reaction.users.remove(fullUser);
+				return;
+			}
+
+			await updateExistingPosts(fullReaction, fullMessage);
+			await addNewPosts(fullReaction, fullMessage);
+		}
```

Comment:
Early return means less Christmas Tree:

```suggestion
		if (!reactboardExists) return; // abort if no reactboard

		if (fullMessage.author.bot) {
			await fullMessage.channel.send(
				`${user.toString()}, you can't use that react on bot messages!`
			);
			await reaction.users.remove(fullUser);
			return;
		}

		if (fullMessage.author.id === user.id) {
			await fullMessage.channel.send(
				`${user.toString()}, you can't use that react your own messages!`
			);
			await reaction.users.remove(fullUser);
			return;
		}

		await updateExistingPosts(fullReaction, fullMessage);
		await addNewPosts(fullReaction, fullMessage);
```